index.html

```diff
@@ -3,47 +3,294 @@
+      var barWrapper = document.createElement('td');
+      var bar = document.createElement('div');
+      bar.setAttribute('class', 'bar');
+
+      // We set the class to 'bar' so we can use the .bar class we have in our styles. This gives it a
+      // consistent height and a color.
+      var VictimsPercentage = d.Victims / MAX_VICTIMS;
+      bar.setAttribute('style', 'width: ' + (VictimsPercentage * 100) + '%;');
+      // I've broken this operation up into two steps, which is a little different than we did in class.
+      // The end result is the same, though. If 'enrollmentPercentage' equals 0.75, this set the style
+      // attribute of our DIV so that it looks something like <div style="width: 75%;"></div>
+      barWrapper.appendChild(bar);
+      row.appendChild(barWrapper);
+      tbody.appendChild(row);
+    });
+  }
+  // The function declaration above just creates the function, it doesn't yet run it. That's what we
+  // do here, to make sure that the table is populated prior to the user clicking anything.
+ 
+  updateTable();
+  // This is creating an event listener, a function that the browser will call every time the specified
+  // event is "fired" by the selected element (here the button with the id 'sort-asc').
+  document.getElementById('sort-asc').addEventListener('click', function () {
+    // Disable the sort if 'asc' was previously selected, otherwise set the sort to 'asc'
+    if (options.sorted === 'asc') {
+      options.sorted = false;
+    } else {
+      options.sorted = 'asc';
+    }
+    // And now we need to re-run the function that populates the table. Note that the results will be
+    // different because we've change the options object.
+    updateTable();
+  });
+  document.getElementById('sort-desc').addEventListener('click', function () {
+    // Disable the sort if 'desc' was previously selected, otherwise set the sort to 'desc'
+    if (options.sorted === 'desc') {
+      options.sorted = false;
+    } else {
+      options.sorted = 'desc';
+    }
+    updateTable();
+  });
+  // This is where we'll need to create new event listeners for our filter and next/previous page buttons.
+        document.getElementById('filter-r1').addEventListener('click', function () {
+      if (options.filtered === 'Region 1') {
+        data = conflict.filter(function (d) { return d.Region === "1" })
+    document.getElementById('filter-r2').addEventListener('click', function () {
+    } else if (options.filtered === 'Region 2') {
+      data = conflict.filter(function (d) { 
+        return d.Region === "2"})
+    } else if (options.filtered === 'Region 3') {
+      data = conflict.filter(function (d) { 
+        return d.Region === "3"})
+    } else if (options.filtered === 'Region 4') {
+      data = conflict.filter(function (d) { 
+        return d.Region === "4"})
+    } else if (options.filtered === 'Region 5') {
+      data = conflict.filter(function (d) { 
+        return d.Region === "5"}
```

Comment:
Maybe this is code in progress; looks like you were transforming the code from above.

Remember that you can use the Console in your Developer Tools (in the View menu, go to Developer) to find what line Javascript's stumbling on.